__tests__/app.test.js

```diff
@@ -111,4 +112,192 @@ describe('GET requests', () => {
+            .then(({ body }) => {
+                expect(body.msg).toBe('Please provide comment body.')
+            })
+        })
+        test('400 - comment username not provided', () => {
+            const malformed2 = {body: 'Great stuff'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(malformed2)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe('Please provide username.')
+            })
+        })
+
+    })
+});
+
+describe('PATCH', () => {
+
+    describe.only('PATCH /api/articles/:article_id', () => {
+        test('201 - responds with the article votes updated correctly', () => {
+            const patchInput = { inc_votes : 20 };
+
+            return request(app)
+                .patch('/api/articles/6')
+                .send(patchInput)
+                .expect(201)
+                .then(({body}) => {
+                    const articleObj = {
+                            article_id: 6,
+                            title: 'A',
+                            topic: 'mitch',
+                            author: 'icellusedkars',
+                            body: 'Delicious tin of cat food',
+                            created_at: '2020-10-18T01:00:00.000Z',
+                            votes: 20,
+                            article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                    }
+                    expect(body.article).toMatchObject(articleObj)
+                })
+        });
+        test('201 - decrements votes for the article correctly', () => {
+            const patchInput = { inc_votes : -20 };
+
+            return request(app)
+                .patch('/api/articles/6')
+                .send(patchInput)
+                .expect(201)
+                .then(({body}) => {
+                    const articleObj = {
+                            article_id: 6,
+                            title: 'A',
+                            topic: 'mitch',
+                            author: 'icellusedkars',
+                            body: 'Delicious tin of cat food',
+                            created_at: '2020-10-18T01:00:00.000Z',
+                            votes: -20,
+                            article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                    }
+                    expect(body.article).toMatchObject(articleObj)
+                })
+        });
+        test('404 - article_id not found', () => {
+            const patchInput2 = { inc_votes : 20 };
+            return request(app)
+                .patch('/api/articles/999')
+                .send(patchInput2)
+                .expect(404)
+                .then(({body}) => {
+                    expect(body.msg).toBe('No articles found for article_id: 999')
+                })
+        });
```

Comment:
We should a 400 for an invalid id e.g. /api/articles/cat

Reply:
added